gcm/client.go

```diff
@@ -0,0 +1,88 @@
+package gcm
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"time"
+)
+
+// gcm Client const variables
+const (
+	GCMClientTimeout = time.Second * 10
+	GCMEndpoint      = "https://fcm.googleapis.com/fcm/send"
+)
+
+// Client is GCM client
+type Client struct {
+	apiKey string
+	Client *http.Client
+}
+
+// Send sends notifications to gcm (TODO: send retry)
+func (gc *Client) Send(p Payload) (*Response, error) {
+	req, err := gc.NewRequest(p)
+	if err != nil {
+		return nil, err
+	}
+
+	res, err := gc.Client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer res.Body.Close()
+
+	resp := &Response{}
+	dec := json.NewDecoder(res.Body)
+	if err = dec.Decode(&resp.Body); err != nil {
+		return nil, err
+	}
+	resp.Header.StatusCode = res.StatusCode
+
+	if res.StatusCode != http.StatusOK {
+		errCode := resp.Body.Error
```

Comment:
https://github.com/kayac/Gunfish/pull/12/files#r101465397

Almost yes.
I used `stringer` to convert Const name into string with `.String()`, so we can compare `Error` message  with `(GCMErrorResponseCode something).String()`.

But, I just realized that sometimes resp.Body.Error is empty (using `registration_ids` instead of `to`), so this cannot set error cod perfectly.

I'll modify that ErrorResponse can have multiple error responses.